util/parsec_project.py

```
@@ -181,38 +200,37 @@ def get_functions_in_topological_order(self, reverse_order: bool = False) -> lis
-        try:
-            cmd = f"{parsec_executable} --rename-main=false --add-instr=false {file_path}"
-            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
-            if result.returncode != 0:
-                msg = f"Error running parsec: {result.stderr} {result.stdout}"
-                raise Exception(msg)
-        except subprocess.CalledProcessError as e:
-            raise Exception("Error running parsec.") from e
+        cmd = f"parsec --rename-main=false --add-instr=false {
+            path if run_mode == ParsecRunMode.FILE else f'-p {path}'
+        }"
+        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.14.14)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)